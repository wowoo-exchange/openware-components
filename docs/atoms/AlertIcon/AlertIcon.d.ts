import * as React from 'react';
import { ErrorIcon } from './ErrorIcon';
import { SuccessIcon } from './SuccessIcon';
interface AlertIconProps {
    /**
     * Type of icon
     */
    type: 'success' | 'error';
}
declare const AlertIcon: React.FunctionComponent<AlertIconProps>;
export { AlertIcon, AlertIconProps, ErrorIcon, SuccessIcon, };
