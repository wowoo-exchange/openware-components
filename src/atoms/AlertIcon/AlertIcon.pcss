.cr-alert-icon {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  min-width: var(--alert-icon-size);
  border-radius: 0 3px 3px 0;
  left: 0;
  position: absolute;

  &--success {
    background-color: var(--open-orders-order-buy);
  }

  &--error {
    background-color: var(--open-orders-order-sell);
  }
}
