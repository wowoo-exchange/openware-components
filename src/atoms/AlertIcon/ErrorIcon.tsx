import * as React from 'react';


export const ErrorIcon = () => {
    return (
        <div className="cr-alert-icon cr-alert-icon--error">
            <svg width="28" height="28" viewBox="0 0 28 28" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path d="M21 7L7.00002 21" stroke="white"/>
                <path d="M7 7L21 21" stroke="white"/>
            </svg>
        </div>
    );
};
