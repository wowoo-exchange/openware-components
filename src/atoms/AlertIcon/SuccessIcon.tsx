import * as React from 'react';


export const SuccessIcon = () => {
    return (
        <div className="cr-alert-icon cr-alert-icon--success">
            <svg width="28" height="28" viewBox="0 0 28 28" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path fillRule="evenodd" clipRule="evenodd" d="M3.64648 12.98L4.35359 12.2729L11.6174 19.5367L22.8904 8.26373L23.5975 8.97083L11.6174 20.9509L3.64648 12.98Z" fill="white"/>
            </svg>
        </div>
    );
};
