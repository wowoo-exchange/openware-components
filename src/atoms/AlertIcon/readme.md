```js
<AlertIcon type="success" />
<AlertIcon type="error" />
```