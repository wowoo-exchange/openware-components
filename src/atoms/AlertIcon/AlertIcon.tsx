import * as React from 'react';
import { ErrorIcon } from './ErrorIcon';
import { SuccessIcon } from './SuccessIcon';

interface AlertIconProps {
    /**
     * Type of icon
     */
    type: 'success' | 'error';
}

const AlertIcon: React.FunctionComponent<AlertIconProps> = (props: AlertIconProps) => {
    const { type } = props;

    return type === 'success'
        ? <SuccessIcon />
        : <ErrorIcon />;
};

export {
    AlertIcon,
    AlertIconProps,
    ErrorIcon,
    SuccessIcon,
};
