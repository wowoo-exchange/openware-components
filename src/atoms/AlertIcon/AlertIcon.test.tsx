import { mount } from 'enzyme';
import * as React from 'react';
import { AlertIcon } from './AlertIcon';

describe('AlertIcon', () => {
    it('should render success icon', () => {
        const wrapper = mount(<AlertIcon type="success" />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveLength(1);
        wrapper.unmount();
    });

    it('should render error icon', () => {
        const wrapper = mount(<AlertIcon type="error" />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveLength(1);
        wrapper.unmount();
    });

    it('should have success className', () => {
        const wrapper = mount(<AlertIcon type="success" />);
        const className = wrapper.find('div').props().className;
        expect(className).toContain('cr-alert-icon cr-alert-icon--success');
        wrapper.unmount();
    });

    it('should have error className', () => {
        const wrapper = mount(<AlertIcon type="error" />);
        const className = wrapper.find('div').props().className;
        expect(className).toContain('cr-alert-icon cr-alert-icon--error');
        wrapper.unmount();
    });
});
