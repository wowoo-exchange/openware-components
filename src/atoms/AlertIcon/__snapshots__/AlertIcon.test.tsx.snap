// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertIcon should render error icon 1`] = `
<AlertIcon
  type="error"
>
  <Component>
    <div
      className="cr-alert-icon cr-alert-icon--error"
    >
      <svg
        fill="none"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 7L7.00002 21"
          stroke="white"
        />
        <path
          d="M7 7L21 21"
          stroke="white"
        />
      </svg>
    </div>
  </Component>
</AlertIcon>
`;

exports[`AlertIcon should render success icon 1`] = `
<AlertIcon
  type="success"
>
  <Component>
    <div
      className="cr-alert-icon cr-alert-icon--success"
    >
      <svg
        fill="none"
        height="28"
        viewBox="0 0 28 28"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.64648 12.98L4.35359 12.2729L11.6174 19.5367L22.8904 8.26373L23.5975 8.97083L11.6174 20.9509L3.64648 12.98Z"
          fill="white"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </Component>
</AlertIcon>
`;
